examples/text_labels_and_annotations/text_rotation.py

```diff
@@ -0,0 +1,49 @@
+"""
+===========================
+Text rotation demonstration
+===========================
+
+The way matplotlib does text layout is counter-intuitive to some, so
+this example is designed to make it a little clearer.  The text is
+aligned by it's bounding box (the rectangular box that surrounds the
+ink rectangle).  The order of operations is rotation then
+alignment, rather than alignment then rotation.  Basically, the text
+is centered at your x,y location, rotated around this point, and then
```

Comment:
I think probably not.

It might be worth having an example for the default rotation mode (this example), and an example for the `'anchor'` rotation mode (to replace http://matplotlib.org/devdocs/examples/pylab_examples/demo_text_rotation_mode.html)